cli/src/main/java/com/liferay/blade/cli/command/CreateCommand.java

```diff
@@ -545,4 +588,95 @@ private boolean _isWorkspaceDir(File dir) {
+		BladeCLI bladeCLI = getBladeCLI();
+
+		try (CloseShieldInputStream closeShieldInputStream = new CloseShieldInputStream(bladeCLI.in());
+			BufferedReader reader = new BufferedReader(new InputStreamReader(closeShieldInputStream))) {
+
+			System.err.println("Missing liferay.workspace.product on gradle.properties");
+
+			Map<String, String> possibleWorkspaceValue = _createPossibleWorkspaceValue(
+				BladeUtil.getWorkspaceProductKeys(true));
+
+			String message =
+				"Please select the Liferay product to target. Typing \"more\" will show all product versions." +
+					_getMessageFromPossibleValues(possibleWorkspaceValue);
+
+			String productKey = _promptSelectLiferayWorkspace(possibleWorkspaceValue, message, reader, bladeCLI.out());
+
+			Map<String, Object> productInfoMap = BladeUtil.getProductInfos();
+
+			ProductInfo productInfo = new ProductInfo((Map<String, String>)productInfoMap.get(productKey));
+
+			BladeUtil.writePropertyValue(new File(dir, "gradle.properties"), "liferay.workspace.product", productKey);
+
+			String formatedTargetPlatformVersion = _formatLiferayVersion(productInfo.getTargetPlatformVersion());
+
+			_verifyCloudDXPWorkspaceV4(formatedTargetPlatformVersion);
+
+			return formatedTargetPlatformVersion;
+		}
+		catch (Exception exception) {
+			return createArgs.getLiferayVersion();
+		}
+	}
+
+	private String _promptSelectLiferayWorkspace(
+		Map<String, String> moreOptionsMap, String message, BufferedReader reader, PrintStream out) {
+
+		String value = Prompter.promptString(message, reader, out);
+
+		while (!moreOptionsMap.containsKey(value) && !moreOptionsMap.containsValue(value) &&
+			   !Objects.equals(value, "more")) {
+
+			System.out.println("Please enter a valid value for liferay version:");
+
+			value = Prompter.promptString("", reader, out);
+		}
+
+		if (moreOptionsMap.containsKey(value)) {
+			value = moreOptionsMap.get(value);
+		}
+		else if (Objects.equals(value, "more")) {
+			Map<String, String> possibleWorkspaceValue = _createPossibleWorkspaceValue(
+				BladeUtil.getWorkspaceProductKeys(false));
+
+			value = _promptSelectLiferayWorkspace(
+				possibleWorkspaceValue, _getMessageFromPossibleValues(possibleWorkspaceValue), reader, out);
+		}
+
+		return value;
+	}
+
+	private void _verifyCloudDXPWorkspaceV4(String chosenProductKey) {
+		try {
+			Properties workspaceProperties = getWorkspaceProperties();
+
+			String property = workspaceProperties.getProperty("liferay.workspace.docker.image.liferay");
+
+			if (StringUtil.isNullOrEmpty(property)) {
+				return;
+			}
+
+			Matcher matcher = _dxpCloudWorkspaceLiferayVersionV4Pattern.matcher(property);
+
+			if (!matcher.find()) {
+				return;
+			}
+
+			String cloudDxpWorkspaceProductVersion = matcher.group();
+
+			if (!cloudDxpWorkspaceProductVersion.equals(chosenProductKey)) {
+				System.out.println(
+					"WARNING: The version of cloud DXP workspace is not match with your selected version.");
```

Comment:
@binhtran92 let's change this from "is not match with your" to "does not match your".